tensorboard/plugins/histogram/histograms_plugin.py

```diff
@@ -84,8 +94,23 @@ def is_active(self):
 
     return bool(self._multiplexer) and any(self.index_impl().values())
 
-  def index_impl(self):
+  def index_impl(self, experiment=None):
     """Return {runName: {tagName: {displayName: ..., description: ...}}}."""
+    if self._data_provider:
+      mapping = self._data_provider.list_tensors(
+          experiment_id=experiment,
```

Comment:
`experiment_id` is never supposed to be `None` (I’m using `=None` as a
hint “consider passing this as a kwarg”, which maybe I shouldn’t?), but
you’re right that there’s an error here, because the distributions
plugin doesn’t thread through the experiment ID correctly. This doesn’t
manifest with the multiplexer provider because it ignores the experiment
ID. Good catch; thank you! Will fix.

Reply:
Okay, I’d missed more places than I thought (some of which mattered, and
some of which didn’t, but have been cleaned up anyway).

As an extra test that the fix threads through the experiment ID
correctly, you can patch the multiplexer data provider to splice in the
experiment ID to the summary description:

```diff
diff --git a/tensorboard/backend/event_processing/data_provider.py b/tensorboard/backend/event_processing/data_provider.py
index 687c4a62..7dad26e2 100644
--- a/tensorboard/backend/event_processing/data_provider.py
+++ b/tensorboard/backend/event_processing/data_provider.py
@@ -99,8 +99,13 @@ class MultiplexerDataProvider(provider.DataProvider):
 
   def list_tensors(self, experiment_id, plugin_name, run_tag_filter=None):
     run_tag_content = self._multiplexer.PluginRunToTagToContent(plugin_name)
+
+    def convert(**kwargs):
+      kwargs["description"] += "...from " + experiment_id
+      return provider.TensorTimeSeries(**kwargs)
+
     return self._list(
-        provider.TensorTimeSeries, run_tag_content, run_tag_filter
+        convert, run_tag_content, run_tag_filter
     )
 
   def read_tensors(
```

…and then verify that these show up in both dashboards:

![Screenshot of histograms dashboard with hacked summary description][ss]

Thanks again for the catch!

[ss]: https://user-images.githubusercontent.com/4317806/70021452-a0079680-1545-11ea-9154-981894df840e.png

Reply:
(Oh, and I made `experiment` non-optional as an extra check.)

Reply:
I am glad I caught something without even trying :)